lib/portlayer/network/context.go

```diff
@@ -191,23 +215,27 @@ func (c *Context) newBridgeScope(id, name string, subnet *net.IPNet, gateway net
 }
 
 func (c *Context) newExternalScope(id, name string, subnet *net.IPNet, gateway net.IP, dns []net.IP, ipam *IPAM) (*Scope, error) {
-	// have to specify IPAM
-	if ipam == nil || len(ipam.pools) == 0 {
-		return nil, fmt.Errorf("no ipam spec for external network")
+	// ipam cannot be specified without gateway and subnet
+	if ipam != nil && len(ipam.pools) > 0 {
+		if isUnspecifiedSubnet(subnet) || gateway.IsUnspecified() {
+			return nil, fmt.Errorf("ipam cannot be specified without gateway and subnet for external network")
+		}
 	}
 
-	if isUnspecifiedSubnet(subnet) || gateway.IsUnspecified() {
-		return nil, fmt.Errorf("neither subnet nor gateway specified for external network")
+	if !isUnspecifiedSubnet(subnet) {
+		// cannot overlap with the default bridge pool
```

Comment:
See https://github.com/hmahmood/vic/blob/3a4a029784c94a94d3b5bbfcad8f339722f65fed/lib/portlayer/network/context.go#L245, which is called from `newScopeCommon`.